### PR TITLE
add mesos slave monitor/statics api

### PR DESCRIPTION
**Pull request by jcmartins**
### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

**Comment by jcmartins** [comment]
Is there anything more to be added? 
How can I help you ?

**Comment by danielnelson** [comment]
I won't have time to review this until next month, I know we need to look into the problems this code caused when it was originally added and see if we can provide the option to get these metrics without causing issues with the database.  This code was disabled in https://github.com/influxdata/telegraf/pull/1686

File: plugins/inputs/mesos/mesos.go [hunk 1]
```diff
@@ -63,7 +65,7 @@ var sampleConfig = `
   #   "system",
   #   "executors",
   #   "tasks",
-  #   "messages",
+  #   "messages",
```
Comment (glinton) [inline plugins/inputs/mesos/mesos.go]: Remove trailing space from this line

File: plugins/inputs/mesos/mesos.go [hunk 2]
```diff
@@ -367,7 +369,7 @@ func getMetrics(role Role, group string) []string {
 	ret, ok := m[group]
 
 	if !ok {
-		log.Printf("I! [mesos] Unknown %s metrics group: %s\n", role, group)
+		log.Printf("I! [mesos] Unkown %s metrics group: %s\n", role, group)
```
Comment (glinton) [inline plugins/inputs/mesos/mesos.go]: Revert back to proper spelling of `Unknown`

File: plugins/inputs/mesos/mesos.go [hunk 3]
```diff
@@ -450,10 +453,9 @@ func (m *Mesos) gatherSlaveTaskMetrics(address string, defaultPort string, acc t
 			return err
 		}
 
+		jf.Fields["framework_id"] = task.FrameworkID
 		timestamp := time.Unix(int64(jf.Fields["timestamp"].(float64)), 0)
-		jf.Fields["executor_id"] = task.ExecutorID
-
-		acc.AddFields("mesos_tasks", jf.Fields, tags, timestamp)
+		acc.AddFields("mesos_monitor_statics", jf.Fields, tags, timestamp)
```
Comment (glinton) [inline plugins/inputs/mesos/mesos.go]: What's your reasoning for changing the measurement name from `mesos_tasks` to `mesos_monitor_statistics`?

File: plugins/inputs/mesos/mesos_test.go [hunk 4]
```diff
@@ -326,7 +329,7 @@ func TestMesosSlave(t *testing.T) {
 	m := Mesos{
 		Masters: []string{},
 		Slaves:  []string{slaveTestServer.Listener.Addr().String()},
-		// SlaveTasks: true,
+		//SlaveMonitorStatics: true,
```
Comment (glinton) [inline plugins/inputs/mesos/mesos_test.go]: Was this meant to be uncommented?

**Comment by jcmartins** [comment]
Why not parse all information about mesos/state?? 
It's have all information about mesos ( master, slaves and framework)

**Comment by glinton** [comment]
As Daniel mentioned, the code to parse all information about mesos/state was commented out because it seemed to be causing problems. I reviewed the PR though so you can address those comments and update your branch and we'll be that much closer to merging if the issues become less.

**Comment by rdelval** [comment]
@glinton or @danielnelson  sorry to bring up an old stale PR but is there any technical problems with re-enabling per task granularity for Mesos now that TSI exists?

**Comment by danielnelson** [comment]
I'm okay with this going forward, we should probably just put a warning in the README like we did for `dcos` and other high cardinality inputs.

**Comment by russorat** [comment]
@jcmartins please let us know when you've addressed the requested changes and fixed the failing checks and conflicts.

**Comment by sjwang90** [comment]
@jcmartins Checking in if this is something you'd like to finish or to see if anyone else would like to take this over, thanks.

**Comment by sjwang90** [comment]
Closing this issue due to inactivity.

If you would still like this plugin, please submit it as an [external plugin](https://github.com/influxdata/telegraf/blob/master/docs/EXTERNAL_PLUGINS.md) that can be used with `execd` to run seamlessly with Telegraf.

**Pull request closed without merging**